File: software/examples/csharp/ExampleThreshold.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Moisture Bricklet

	// Callback function for moisture value reached callback
	static void MoistureReachedCB(BrickletMoisture sender, int moisture)
	{
		Console.WriteLine("Moisture Value: " + moisture);
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletMoisture m = new BrickletMoisture(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get threshold callbacks with a debounce time of 1 second (1000ms)
		m.SetDebouncePeriod(1000);

		// Register moisture value reached callback to function MoistureReachedCB
		m.MoistureReached += MoistureReachedCB;

		// Configure threshold for moisture value "greater than 200"
		m.SetMoistureCallbackThreshold('>', 200, 0);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/java/ExampleThreshold.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletMoisture;

public class ExampleThreshold {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;
	private static final String UID = "XYZ"; // Change to your UID

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions you
	//       might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletMoisture m = new BrickletMoisture(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get threshold callbacks with a debounce time of 1 second (1000ms)
		m.setDebouncePeriod(1000);

		// Configure threshold for "greater than 200"
		m.setMoistureCallbackThreshold('>', 200, 0);

		// Add threshold reached listener for moisture value greater than 200
		m.addMoistureReachedListener(new BrickletMoisture.MoistureReachedListener() {
			public void moistureReached(int moisture) {
				System.out.println("Moisture Value: " + moisture);
			}
		});

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}
